compiler/src/dmd/lexer.d

```diff
@@ -3072,6 +3088,66 @@ class Lexer
         return u;
     }
 
+    version (DMDLIB)
+    {
+        /***************************************************
+     * Parse a comment embedded between t.ptr and p.
+     * Remove trailing blanks and tabs from lines.
+     * Replace all newlines with \n.
+     * Preserve leading comment string in each line.
+     * Append to previous one for this token.
+     */
+        private void getRawComment(Token* t) pure
+        {
+            const(char)* q = t.ptr; // start of comment text
+            OutBuffer buf;
+
+            for (; q < p /* start of next token */ ; q++)
+            {
+                char c = *q;
```

Comment:
Can this stuff not be reused from the existing comment lexing stuff?

Reply:
The existing function strips the comment prefix, and stores only the string. There's no way for us to figure out whether a comment is `//`, `/**/`, `///`, or any of the other types. I couldn't see a way for us to store that information apart from using a different function to lex the comment and store it as a raw string, prefix included.

Reply:
if you want to take inspiration from libdparse, it stores the raw tokens (as slice of tokens) for comments and whitespace and has a getter function that extracts the ddoc comments and strips them of the borders (once, then memoizes the result)

Reply:
SDC keeps comments as a token.

For here I would get all the comment lexing logic into one place and then return the info anything else would need rather than just having basically the same logic twice.

Reply:
@WebFreak001 what do you mean by slice of tokens (wrt comments)

Reply:
An important consideration for a lexer is raw speed. It's better to pay the price in extra code and have specific lexers hand-tuned for exactly what it needs, rather than have more generic code with multiple uses.

Reply:
Perhaps, but classifying (say) a comment after when it has been lexed (in all but name) is not where slowness comes from. 

SDC (as mentioned above), does lots of SIMD/SWAR tricks that make it very fast at lexing, it's still all pretty much generic — the building blocks are the same everywhere, and just complicated enough that having them done 5 times over would preclude make all 5 versions faster.